src/lib/customer.ts

```diff
@@ -1,5 +1,37 @@
 const CIO_BASE_URL = `https://beta-api.customer.io/v1/api/`
 
+async function fetchCustomer(cioId: string) {
+  return new Promise(async (resolve, reject) => {
+    const cioApiPath = `/customers/${cioId}/attributes`
+    const headers = new Headers({
+      Authorization: `Bearer ${process.env.CUSTOMER_IO_APPLICATION_API_KEY}`,
+    })
+
+    // if CIO isn't responding in 4s we want to fallback and show the page
+    const TIMEOUT = 4000
```

Comment:
had somebody mention that the homepage timed out for them and I assume it's a connection issue with CIO and made this to give a minimum time and bail if it fails to load. It was the only unknown async and could definitely affect folks in remote locales.